### ISSUE TITLE
Obsolete modules reported at application launch

### ISSUE THREAD

**Issue by Zesaric** [EmilyBjoerk/lsml#683]
**LSML version:** 2.0.7
**Screen resolution:** 1680x1050 (3x)
**Java version:** 8u112

**Steps to reproduce issue:**
1. Open LSML with a saved garage containing loadouts with obsolete modules.

**Actual result:**
Each Loadout that contains invalid modules displays a separate error message. As I have approx. 300 Mechs all of which are re-build in LSML, I had to click "OK" 300 times before I finally could use LSML... ;-)
Not a big deal, but maybe it would a "smoother" start for 2.0.7 to just ignore obsolete modules instead of reporting them as errors.

**Expected result:**
LSML should have started without error messages! :-)

**Comment by EmilyBjoerk** [comment]
It's a difficult call to make. Out of principle I don't want to modify the user's data without telling them, yet I realise it's a bit annoying when upgrading from an older garage. I'll see what I can do. Would you be willing to give me your old garage file to test with? You did make a backup, right? ;)

**Comment by Zesaric** [comment]
Actually my idea was to not modify the garage file but rather set some exception as to when an error is reported. You could do something like "If missing module ID equals <obsolete module IDs> then suppress error message".

 

However, it's not a big problem and even if I have many Mechs I was was able to click through the errors in a matter of seconds. And it was only once! :)

 

So my suggestion is not an essential improvment but rather a cosmetic one. And I actually just reported it because I don't know how easily an exception as described above could be implemented. So if you still want to tackle this, I'll happily send you my garage file - but if you want to postpone or close the issue it's also ok with me. Just let me know your preference.

 

 

Von: Emily Björk [mailto:notifications@github.com] 
Gesendet: Montag, 17. Juli 2017 20:07
An: EmilyBjoerk/lsml <lsml@noreply.github.com>
Cc: Zesaric <zesaric@gmx.net>; Author <author@noreply.github.com>
Betreff: Re: [EmilyBjoerk/lsml] Obsolete modules reported at application launch (#683)

 

It's a difficult call to make. Out of principle I don't want to modify the user's data without telling them, yet I realise it's a bit annoying when upgrading from an older garage. I'll see what I can do. Would you be willing to give me your old garage file to test with? You did make a backup, right? ;)

**Comment by EmilyBjoerk** [comment]
Yeah, I'm not going to hand list all the obsolete module IDs that's too much work. I can see if I can collect all the errors into one big error list instead and show that. But I'd need a decent garage file to work with for that.

**Comment by Zesaric** [comment]
Not sure if my Garage file is a decent one ;-) but in any case I'll send it to you as a sample to work on.
